readme.txt
=== Juliet ===

Author: LyraThemes www.lyrathemes.com
Requires at least: WordPress 4.5
Tested up to: WordPress 4.7+
Version: 1.3.1
License: GPLv3 or later
License URI: http://www.gnu.org/licenses/gpl-3.0.en.html
Tags: one-column, two-columns, right-sidebar, flexible-header, custom-background, custom-header, custom-menu, featured-images, footer-widgets, sticky-post, theme-options, threaded-comments, translation-ready, blog, food-and-drink, entertainment, photography


== Description ==

Juliet is an elegant, blog-focused, feminine theme perfect for lifestyle blogging. The front page allows you to showcase featured posts. Juliet is the perfect blogging theme with a wide range of features for customizing the layout.

* Off Canvas Menus
* Fancy Scrollbar
* Front Page Banner
* Front Page Featured Posts		
* Image and Text Logos		
* Background Image		
* Background Color	
* Special Sticky Posts Format with Signature	
* Optional Full Width Posts and Pages		
* Excerpt or Full Blog Feed Layouts
* Integration with Recent Posts Widget With Thumbnails
* Integration with MailPoet
* Easy to Create Social Media Icon Menus
* Header and Footer Area Widgets
* Extra Page, Post and Front Page Specific Sidebars		
* Footer Widgets
* Full Width Footer Widget Area


For a complete list of features, go to  https://www.lyrathemes.com/juliet/
For a detailed documentation, please go to https://www.lyrathemes.com/documentation/juliet.pdf
For support: support@lyrathemes.com or https://www.lyrathemes.com/support

== UPGRADE TO PRO FOR THESE AWESOME FEATURES ==

https://www.lyrathemes.com/juliet-pro/

* Fixed Header/Navigation
* Front Page Posts Slider
* Front Page Custom Slider
* Strip Slider (3 Slides at a Time)
* Simple Slider (1 Slide at a Time)
* Fron Page Featured Links/Boxes 
* Front Page Highlight Post		
* Front Page Focus Posts
* Integration with WP Instagram Widget	
* Special Lookbook / Index Template		
* Built-in Social Media Sharing Icons (Using JetPack)		
* Built-in Ads
* Built-in Related Posts		
* Unlimited Color Choices		
* 600+ Google Fonts		
* Google Analytics		

== Copyright ==

Juliet Pro, Copyright 2016 Lyrathemes.com
It is distributed under the terms of the GNU GPL

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.

The theme uses the following third-party resources:

* HTML5 Shiv v3.7.0 (c) Alexander Farkas. MIT/GPL2. https://github.com/aFarkas/html5shiv
* Respond.js v1.3.0 (c) Scott Jehl. MIT/GPLv2. https://github.com/scottjehl/Respond
* Bootstrap v3.3.5 (c) Twitter Inc. MIT. http://getbootstrap.com
* Kirki v. GPLv2 or later https://aristath.github.io/kirki/
* TGM-Plugin-Activation v2.6.1 (c) Thomas Griffin. GPL-2.0+. http://tgmpluginactivation.com/
* Font Awesome 4.7 by Dave Gandy. MIT. http://fontawesome.io
* wp-bootstrap-navwalker (c) Edward McIntyre. GPL-2.0. https://github.com/twittem/wp-bootstrap-navwalker
* Sample preview images in /sample/images are Creative Commons CC0. Details below.
* Glyphicon Halflings (c) http://glyphicons.com/license/. MIT.
* jquery.nicescroll 3.6.0 (c) InuYaksa. MIT. http://nicescroll.areaaperta.com
* slick 1.6.0 (c) Ken Wheeler. MIT. https://github.com/kenwheeler/slick

== Sample Images ==

featured1.jpg	https://pixabay.com/en/desk-office-desk-interior-notebook-1082044/
featured2.jpg	https://unsplash.com/photos/GwNsgnSAfQM
featured3.jpg	https://pixabay.com/en/book-read-abstract-literature-2208046/
featured4.jpg	https://unsplash.com/photos/RTXwyy3ctXE

thumb1.jpg	https://unsplash.com/photos/BmsmOjGQegU 
thumb2.jpg	https://unsplash.com/photos/4BHcRGCDXQI
thumb3.jpg	https://unsplash.com/new?photo=9Jq5UvyXr8A 
thumb4.jpg	https://unsplash.com/new?photo=9Jq5UvyXr8A 

header.jpg	https://unsplash.com/photos/GJMlSBS0FhU


changelog.txt
= v 1.3.1 
* 6/8/2017
* Removed the shortcut for arrays from WooCommerce code
* Fixed: Horizontal scrollbar
* Fixed: Logo is not h1 on internal pages
* Updated: Kirki to 2.3.8

= v 1.3 [MAJOR UPDATE]
* 5/30/2017 
* Complete WooCommerce integration added (shop, cart, product, checkout pages)
* Fixed 4th featured post on home page
* Caption box on banner is now transparent (Classic)
* Caption box not displayed if no heading/description is supplied in banner

= v 1.2.4
* 5/19/2017
* Fancy scrollbar off by default

= v 1.2.3
* 5/18/2017
* juliet_custom_colors defined in theme-defaults.php
* 4th featured post fixed
* Updated screenshot
* Updated wp_bootstrap_navwalker to show item title
* Fixed missing entry-title, author, updated errors (https://www.wpcub.com/fix-missing-required-field-entry-titleupdatehcard-error-in-wordpress/)
* ADDED: CSS rules to automatically make Vimeo and YouTube videos responsive

= v 1.2.2
* 5/13/2017
* Added juliet_custom_colors_scrollbar
* Small fix to sidebar (missing div)
* Reworked juliet_skin variable in function.php
* add_theme_support( 'woocommerce' );

= v 1.2.1
* woocommerce.php fixed.
* '404' is now translatable.
* Using individual variables for each front page featured post (not global $post)

= v 1.2 [MAJOR UPDATE]
* 04/25/2017
* Updated screenshot
* Moved "Colors" above "Background Image", renamed "Custom Colors"
* Updated font-awesome to 4.7
* Fixed escaping throughout all files
* Updated thumbnail size to 700x600
* Added non minified versions of JS files
* Removed theme update checker
* Updated language file
* Banner caption allows HTML now
* Updated sample images
* Changed structure of off canvas menu 
* Removed <span>s from widget headings
* NEW: Added WooCommerce template and styling
* NEW: Added Contemporary skin
* NEW: Added banner overlay color selection
* NEW: Added example content check

= v 1.1
* 03/12/2017
* Updated screenshot
* Custom CSS only shown for WP < 4.7
* Removed juliet_title() and now using juliet_archive_title() for get_the_archive_title()
* Removed juliet_get_random_post()
* Updated frontpage-featured.php to add checks for featured posts variables
* Font Awesome updated to version 4.7 (?)
* Added formatting for galleries
* Theme update checker added

= v 1.0 =
* 08/13/2016 
* Initial Release

functions.php
<?php
/**
 * Juliet functions and definitions
 *
 * @package juliet
 */
?>
<?php

/*------------------------------
 Customizer
 ------------------------------*/
 
if ( ! class_exists( 'Kirki' ) ) {
    include_once( dirname( __FILE__ ) . '/inc/kirki/kirki.php' );
}
require get_template_directory() . '/customize/theme-defaults.php';
require get_template_directory() . '/customize/kirki-config.php';
require get_template_directory() . '/customize/customizer.php';

function juliet_customize_register( $wp_customize ) {
    $wp_customize->get_section('colors')->title = esc_html__( 'Custom Colors', 'juliet' );
    $wp_customize->get_section('colors')->priority = 75;
}
add_action( 'customize_register', 'juliet_customize_register' );

if(is_admin())  add_action( 'customize_controls_enqueue_scripts', 'juliet_custom_customize_enqueue' );
function juliet_custom_customize_enqueue() {
    wp_enqueue_style( 'juliet-customizer', get_template_directory_uri() . '/customize/style.css' );
}

/*------------------------------
 Setup
 ------------------------------*/

function juliet_setup() {
    global $juliet_defaults;
    load_theme_textdomain( 'juliet', get_template_directory() . '/languages' ); 
    
    register_nav_menus( array(  'header'    => esc_html__( 'Main Menu', 'juliet' ), 
                                'offcanvas' => esc_html__( 'Off Canvas Menu', 'juliet') ) );
    
    add_theme_support( 'automatic-feed-links' );
    add_theme_support( 'title-tag' );
    add_theme_support( 'custom-logo', array('height' => 80, 'width' => 250,'flex-height' => true,'flex-width'  => true ) );
    add_theme_support( 'custom-background');
	add_theme_support( 'html5', array( 'comment-form', 'comment-list', 'gallery', 'caption' ) );
    
    $args = array(
        'flex-width'    => true,
        'width'         => 1200,
        'flex-height'    => true,
        'height'        => 625,
        'default-image' => $juliet_defaults['juliet_custom_header'],
    );
    add_theme_support( 'custom-header', $args );

    add_theme_support( 'post-thumbnails' );
    set_post_thumbnail_size(            700, 600, true );
    add_image_size( 'juliet-slider',    1200, 625, true );
    add_image_size( 'juliet-thumbnail', 700, 600, true );
    add_image_size( 'juliet-featured',  540, 540, true );

    add_post_type_support('page', 'excerpt');
	
	#https://make.wordpress.org/core/2016/11/26/extending-the-custom-css-editor/
    if ( function_exists( 'wp_update_custom_css_post' ) ) {
        $css = juliet_get_option('juliet_advanced_css');
        if ( $css ) {
            $core_css = wp_get_custom_css(); 
            $return = wp_update_custom_css_post( $core_css . $css );
            if ( ! is_wp_error( $return ) ) {
                remove_theme_mod( 'juliet_advanced_css' );
            }
        }
    }
	
	if ( class_exists( 'WooCommerce' ) ) {
		add_theme_support( 'wc-product-gallery-zoom' );
		add_theme_support( 'wc-product-gallery-lightbox' );
		add_theme_support( 'wc-product-gallery-slider' );
		add_theme_support( 'woocommerce' );
	}
}
add_action( 'after_setup_theme', 'juliet_setup' );

/*------------------------------
 Styles and Scripts
 ------------------------------*/

function juliet_scripts() {
    
    /* Styles */
    
    //fonts
	$juliet_skin = juliet_get_option('juliet_skin');
	wp_enqueue_style('juliet-googlefont-logo-tagline', '//fonts.googleapis.com/css?family=Crimson+Text:400,400i,600,600i,700,700i'); #logo tagline
	if( $juliet_skin == 'Classic') { 
		wp_enqueue_style('juliet-googlefont-body', '//fonts.googleapis.com/css?family=Lato:400,400i,700,700i'); #body
    } else {
		wp_enqueue_style('juliet-googlefont-body', '//fonts.googleapis.com/css?family=Lato:400,400i,700,700i'); #body
		wp_enqueue_style('juliet-googlefont-headings', '//fonts.googleapis.com/css?family=Montserrat:300,300i,400,400i,500,500i,700,700i'); #headings
	}
    wp_register_style('bootstrap', get_template_directory_uri() . '/assets/css/bootstrap.min.css' );
    wp_register_style('font-awesome', get_template_directory_uri().'/assets/css/font-awesome.min.css' );

    //default stylesheet
    $deps = array('bootstrap', 'font-awesome');
    wp_enqueue_style('juliet-style', get_stylesheet_uri(), $deps );

    if ( class_exists( 'WooCommerce' ) ) {
        wp_enqueue_style('slick-style', get_template_directory_uri().'/assets/css/slick.min.css' );
        wp_enqueue_style('juliet-woocommerce-style', get_template_directory_uri().'/woocommerce/woocommerce.css', array( 'juliet-style' ) );
    }
    
    // Load html5shiv.js
	wp_enqueue_script( 'juliet-html5', get_template_directory_uri() . '/assets/js/html5shiv.js' ); //, array('juliet-style'), '3.7.0' );
	wp_script_add_data( 'juliet-html5', 'conditional', 'lt IE 9' );
    // Load respond.min.js
	wp_enqueue_script( 'juliet-respond', get_template_directory_uri() . '/assets/js/respond.min.js' ); //, array('juliet-style'), '1.3.0' );
	wp_script_add_data( 'juliet-html5', 'conditional', 'lt IE 9' );
    
    /* Scripts */
    
    wp_enqueue_script('bootstrap', get_template_directory_uri().'/assets/js/bootstrap.min.js', array('jquery'), '', true );
    if ( class_exists( 'WooCommerce' ) ) {
        wp_enqueue_script('slick', get_template_directory_uri().'/assets/js/slick.min.js', array('jquery'), '', true);
    }
    if(juliet_get_option('juliet_enable_fancy_scrollbar') == 1){
        wp_enqueue_script('nice-scroll', get_template_directory_uri() . '/assets/js/nicescroll.min.js', array('jquery'), '', true );
        wp_enqueue_script('juliet-js', get_template_directory_uri() . '/assets/js/juliet.js', array('jquery', 'bootstrap', 'nice-scroll'), '', true );
    } else {
        wp_enqueue_script('juliet-js', get_template_directory_uri() . '/assets/js/juliet.js', array('jquery', 'bootstrap'), '', true );
    }
    
    //comments
    if ( is_singular() && comments_open() && get_option( 'thread_comments' ) ) {
        wp_enqueue_script( 'comment-reply' );
    }
    
}
add_action( 'wp_enqueue_scripts', 'juliet_scripts' );

/*------------------------------
 Custom CSS
 ------------------------------*/

if ( ! function_exists( 'juliet_custom_css_banner_overlay' ) ) :
function juliet_custom_css_banner_overlay(){ 
	global $juliet_defaults;
    $juliet_frontpage_banner_overlay_color = juliet_get_option('juliet_frontpage_banner_overlay_color');
	$juliet_frontpage_banner_overlay_show = juliet_get_option('juliet_frontpage_banner_overlay_show');
    if($juliet_frontpage_banner_overlay_show == 0){
		echo "<style>";
		echo ".frontpage-banner:before, .frontpage-slider .item:before{content:none;}";
		echo "</style>";
	} else if($juliet_frontpage_banner_overlay_color != $juliet_defaults['juliet_frontpage_banner_overlay_color']) {
		echo "<style>";
		echo ".frontpage-banner:before{background-color:".esc_attr($juliet_frontpage_banner_overlay_color).";}"; 
		echo ".frontpage-slider .item:before{background-color:".esc_attr($juliet_frontpage_banner_overlay_color).";}"; 
		echo "</style>";
	}
} 
endif;
add_action('wp_head','juliet_custom_css_banner_overlay', 98);

if ( ! function_exists( 'juliet_custom_css' ) ) :
function juliet_custom_css() {
    $juliet_advanced_css = juliet_get_option('juliet_advanced_css');
    if($juliet_advanced_css != '') {    
        echo '<!-- Custom CSS -->';
        $output="<style>" . wp_strip_all_tags($juliet_advanced_css) . "</style>";
        echo $output;
        echo '<!-- /Custom CSS -->';
    }
}
endif;
add_action('wp_head','juliet_custom_css', 99);

/*------------------------------
 Widgets
 ------------------------------*/
require_once get_template_directory() . '/widgets/widgets.php';

/*------------------------------
 Content Width 
 ------------------------------*/
if ( ! isset( $content_width ) ) {
    $content_width = 1200;
}

/*------------------------------
 wp_bootstrap_navwalker
 ------------------------------*/
require_once get_template_directory() . '/inc/wp_bootstrap_navwalker.php';

/*------------------------------
 TGM_Plugin_Activation
 ------------------------------*/

require_once get_template_directory() . '/inc/class-tgm-plugin-activation.php';
add_action( 'tgmpa_register', 'juliet_register_required_plugins' );
function juliet_register_required_plugins() {
	$plugins = array(
        array(
			'name'      => esc_html__('Recent Posts Widget With Thumbnails', 'juliet'),
			'slug'      => 'recent-posts-widget-with-thumbnails',
			'required'  => false,
		)
	);
	$config = array(
		'id'           => 'juliet',                // Unique ID for hashing notices for multiple instances of TGMPA.
		'default_path' => '',                      // Default absolute path to bundled plugins.
		'menu'         => 'tgmpa-install-plugins', // Menu slug.
		'has_notices'  => true,                    // Show admin notices or not.
		'dismissable'  => true,                    // If false, a user cannot dismiss the nag message.
		'dismiss_msg'  => '',                      // If 'dismissable' is false, this message will be output at top of nag.
		'is_automatic' => false,                   // Automatically activate plugins after installation or not.
		'message'      => '',                      // Message to output right before the plugins table.
	);
	tgmpa( $plugins, $config );
}


/*------------------------------
 Helper
 ------------------------------*/

if ( ! function_exists( 'juliet_get_option' ) ) :
function juliet_get_option($key){
    global $juliet_defaults;
    if (array_key_exists($key, $juliet_defaults)) 
        $value = get_theme_mod($key, $juliet_defaults[$key]); 
    else
        $value = get_theme_mod($key);
    return $value;
}
endif;

if ( ! function_exists( 'juliet_get_bootstrap_class' ) ) :
function juliet_get_bootstrap_class($columns){
    switch($columns){
        case 1: return 'col-md-12'; break;
        case 2: return 'col-lg-6 col-md-6 col-sm-6'; break;
        case 3: return 'col-lg-4 col-md-4 col-sm-4 col-xs-12'; break;
        case 4: return 'col-lg-3 col-md-3 col-sm-6 col-xs-12'; break;
        case 5: return 'col-md-20'; break;
        case 6: return 'col-lg-2 col-md-2 col-sm-2 col-xs-6'; break;
    }
}
endif;

if ( ! function_exists( 'juliet_get_sample' ) ) :
function juliet_get_sample($what){
    global $juliet_defaults;
    switch($what){
        case 'juliet-thumbnail':    $images = $juliet_defaults['juliet_thumbnail_sample']; $rand_key = array_rand($images, 1); return ($images[$rand_key]);
        case 'juliet-full':         $images = $juliet_defaults['juliet_full_sample']; $rand_key = array_rand($images, 1); return ($images[$rand_key]);
        case 'juliet-featured':     $images = $juliet_defaults['juliet_featured_sample']; $rand_key = array_rand($images, 1); return ($images[$rand_key]);    
    }
}
endif;

if ( ! function_exists( 'juliet_get_body_class' ) ) :
function juliet_get_body_class(){
    $class = 'boxed';
    return $class;
}
endif;

if ( ! function_exists( 'juliet_show_custom_css_field' ) ) :
function juliet_show_custom_css_field(){
    if(get_bloginfo('version') >= 4.7){
        $juliet_advanced_css = juliet_get_option('juliet_advanced_css');
        if($juliet_advanced_css == '') return false;
        else return true;
    } 
    return true;
}
endif;



/*------------------------------
 Filters
 ------------------------------*/

//https://gist.github.com/hatsumatsu/e5df3b0bf9ac3389c0b0
function juliet_embed_oembed_html( $html, $url, $attributes, $id ) {
    $class = '';
    if( strpos( $url, 'youtu' ) || strpos( $url, 'vimeo' ) ) {
        $class .= ' iframe-video';
    }
    return '<div class="' . esc_attr( $class ) . '">' . $html . '</div>';
}
add_filter( 'embed_oembed_html', 'juliet_embed_oembed_html', 99, 4 );

#disable comments on media attachments
function juliet_filter_media_comment_status( $open, $post_id ) {
	$post = get_post( $post_id );
	if( $post->post_type == 'attachment' ) {
		return false;
	}
	return $open;
}
add_filter( 'comments_open', 'juliet_filter_media_comment_status', 10 , 2 );

#move comment field to the bottom of the comments form
function juliet_move_comment_field_to_bottom( $fields ) {
    $comment_field = $fields['comment'];
    unset( $fields['comment'] );
    $fields['comment'] = $comment_field;
    return $fields;
}
add_filter( 'comment_form_fields', 'juliet_move_comment_field_to_bottom' );

#excerpt length
function juliet_excerpt_length( $length ) {
	return 45;
}
add_filter( 'excerpt_length', 'juliet_excerpt_length', 999 );

/*
#add class to page nav
function juliet_wp_page_menu_class( $class ) {
  return preg_replace( '/<ul>/', '<ul class="nav navbar-nav">', $class, 1 );
}
add_filter( 'wp_page_menu', 'juliet_wp_page_menu_class' );
*/

#nicescroll
function juliet_nice_scroll(){
    if(juliet_get_option('juliet_enable_fancy_scrollbar') != 1) return;
    global $juliet_defaults;
    $juliet_custom_colors = juliet_get_option('juliet_custom_colors');
    if($juliet_custom_colors == 0) $color = $juliet_defaults['juliet_custom_colors_scrollbar'];
    else $color = juliet_get_option('juliet_custom_colors_scrollbar');
    $script = 'jQuery(document).ready(function($){jQuery("html").niceScroll({ cursorcolor:"' . esc_attr($color) . '",
                                cursorborder:"' . esc_attr($color) . '",
                                cursoropacitymin:0.2,
                                cursorwidth:10,
                                zindex:10,
                                scrollspeed:60,
                                mousescrollstep:40}); });';
    wp_add_inline_script( 'juliet-js', $script );
}
add_action( 'wp_enqueue_scripts', 'juliet_nice_scroll', 998 );

#get_the_archive_title
function juliet_archive_title( $title ) {
    if( is_home() && get_option('page_for_posts') ) {
        $title = get_page( get_option('page_for_posts') )->post_title;
    }
    else if( is_home() ) {
        $title = juliet_get_option('juliet_blog_feed_label');
        $title = esc_html($title);
    }  
    return $title;
}
add_filter( 'get_the_archive_title', 'juliet_archive_title' );


/*------------------------------
 Example / Default
 ------------------------------*/
 
#default nav top level pages
function juliet_default_nav(){
	echo '<div class="navbar-collapse collapse">';
    echo '<ul class="nav navbar-nav">';
    $pages = get_pages();  
    $n = count($pages); 
    $i=0;
    foreach ( $pages as $page ) {
        $menu_name = $page->post_title;
        $menu_link = get_page_link( $page->ID );
        if(get_the_ID() == $page->ID) $current_class = "current_page_item";
        else { $current_class = ''; $home_current_class = ''; }
        $menu_class = "page-item-" . $page->ID;
        echo "<li class='page_item ".esc_attr($menu_class)." $current_class'><a href='".esc_url($menu_link)."'>".esc_html($menu_name)."</a></li>";
        $i++;
    } 
    echo '</ul>';
	echo '</div>';
}


function juliet_example_footer_widgets(){
    
    echo '<div class="sidebar-footer footer-row-1">';
    echo '<div class="row">';
    
    echo '<div class="col-lg-3 col-md-3 col-sm-6 col-xs-12">';
    the_widget('WP_Widget_Pages', 'title=Pages', 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="footer-row-1-widget widget">&after_widget=</div>');
    echo '</div>';
    
    echo '<div class="col-lg-3 col-md-3 col-sm-6 col-xs-12">';
    the_widget( 'WP_Widget_Archives', 'title=Archives', 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="footer-row-1-widget widget">&after_widget=</div>');
    echo '</div>';
    
    echo '<div class="col-lg-3 col-md-3 col-sm-6 col-xs-12">';
    the_widget('WP_Widget_Recent_Posts', 'title=Recent Posts', 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="footer-row-1-widget widget">&after_widget=</div>');
    echo '</div>';
    
    echo '<div class="col-lg-3 col-md-3 col-sm-6 col-xs-12">';
    the_widget( 'WP_Widget_Categories', 'title=Categories', 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="footer-row-1-widget widget ">&after_widget=</div>');
    echo '</div>';
    
    echo '</div>';
    echo '</div>';
}

#juliet_example_sidebar
function juliet_example_sidebar(){
    echo '<div class="sidebar-default" >';
    the_widget('WP_Widget_Search', 'title=' . esc_html__('Search', 'juliet'), 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="default-widget widget widget_search">&after_widget=</div>');
    the_widget('WP_Widget_Pages', 'title=' . esc_html__('Pages', 'juliet') , 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="default-widget widget">&after_widget=</div>');
    the_widget('WP_Widget_Recent_Posts', 'title=' . esc_html__('Recent Posts', 'juliet') , 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="default-widget widget">&after_widget=</div>');
    the_widget( 'WP_Widget_Archives', 'title=' . esc_html__('Archives', 'juliet'), 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="default-widget widget">&after_widget=</div>');
    the_widget( 'WP_Widget_Categories', 'title=' . esc_html__('Categories', 'juliet'), 'before_title=<h3 class="widget-title">&after_title=</h3>&before_widget=<div class="default-widget widget">&after_widget=</div>');
    echo '</div>';
}

/*------------------------------
 Woocommerce
 ------------------------------*/

 require_once get_template_directory() . '/woocommerce/functions_woocommerce.php';
?>
